pyproject.toml

```diff
@@ -1,22 +1,134 @@
-[tool.black]
-line-length = 79
-target-version = ['py38']
-include = '\.pyi?$'
-exclude = '''
-
-(
-  /(
-      \.eggs         # exclude a few common directories in the
-    | \.git          # root of the project
-    | \.hg
-    | \.mypy_cache
-    | \.tox
-    | \.venv
-    | _build
-    | buck-out
-    | build
-    | dist
-    | etc
-  )/
-)
-'''
+[tool.ruff]
+line-length = 88
+target-version = "py310"
+
+[tool.ruff.format]
+preview = false
+
+[tool.ruff.lint]
+ignore = [
+    # NOTE: Non-permanent exclusions should be added to the ".ruff.toml" file.
+
+    # flake8-commas (COM)
+    # https://docs.astral.sh/ruff/rules/#flake8-commas-com
+    "COM812",  # Trailing comma missing.
```

Comment:
I can't see why this stayed; running pre-commit throws no issues.

I've added in intentional offenses which it has thrown up, so it's definitely not on my end.

Reply:
Looking into this, I can see why you might've done, and I don't think having it here is end of the world. Happy to leave it in